pkg/querier/distributor_queryable_streaming.go

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querier
+
+import (
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+
+	"github.com/grafana/mimir/pkg/ingester/client"
+	"github.com/grafana/mimir/pkg/querier/stats"
+	"github.com/grafana/mimir/pkg/storage/series"
+)
+
+type streamingChunkSeries struct {
+	labels            labels.Labels
+	chunkIteratorFunc chunkIteratorFunc
```

Comment:
Very possibly - `chunkIteratorFunc` will be one of the choices from [`getChunksIteratorFunction`](https://github.com/grafana/mimir/blob/b6566fff9d410f589b58904472fe4912e17e0a8c/pkg/querier/querier.go#L101), is the same for the entire life of the process, and each implementation returns the same kind of iterator each time, so we should be able to avoid allocating a new iterator for each series if we reuse them.

Reply:
(I've added this to our to-do list to come back and investigate once this PR is merged @codesome)